frame/support/procedural/src/pallet/parse/composite.rs

```diff
@@ -0,0 +1,139 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
```

Comment:
This is great for our purposes. I wonder, though, how easy would it be for an outside user to add a custom composite for, say their own parachain? Would they have to patch this file?

Reply:
They'd actually have to do a little more stuff:

1. Modify the parser in `construct_runtime` so that it knows how to scan for the existence of a new aggregate enum via a new pallet part.
2. Instruct `construct_runtime` to check for the new pallet part and expanding it into the aggregate enum, complete with conversion functions as well.
3. Scan for all pallets during the expansion mentioned in 2 for any other pallets that contain the new pallet part, and aggregate all of them into an aggregated enum.